ocaml/utils/warnings.ml

```diff
@@ -34,6 +34,11 @@ type constructor_usage_warning =
   | Not_constructed
   | Only_exported_private
 
+type upstream_compat_warning =
+  | Immediate_erasure of string
```

Comment:
Maybe add tiny examples of each here? I know other declarations don't do this, but it's so so helpful to. :)